### ISSUE TITLE
v0.23.0 causes issue with TTY

### ISSUE THREAD

**Issue by raphaelfff** [charmbracelet/bubbletea#595]
I dont have much details on this yet, but from v0.22.1 and v0.23.0 there seem to be some issue with TTY

Here is the scenario:
bbt runs for a bit, then exits (`.Start()` terminates), then a `exec.NewCommand` runs with os.Stdout & os.Stdin connected directly, that `exec.NewCommand` runs a python script that reads stdout (same thing happens with a `docker exec -it`)

Before v0.23.0 it works fine, on v0.23.0 it exits with some tty error: `lost sys.stdin` (python) or `the input device is not a TTY` (docker)

The shell is bash

**Comment by muesli** [comment]
Just so I understand this correctly: you're running a `tea.Program`, which finishes first, and then you're trying to execute another sub-command afterwards? Could you try to provide a simple example we can reproduce this with? Thank you!

**Comment by raphaelfff** [comment]
Yes, you are correct, i m trying to get a min repro example

**Comment by raphaelfff** [comment]
@muesli This is the smallest repro example:

First run `docker run -d --name testbbt ubuntu sleep 9999`

The following works with 0.22.0, and doesnt with 0.23.0
```
package main

import (
	"fmt"
	"os"
	"os/exec"
	"time"

	tea "github.com/charmbracelet/bubbletea"
)

type model struct {
	ch chan bool
}

func (m model) Init() tea.Cmd {
	return nil
}

func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
	select {
	case <-m.ch:
		return m, tea.Quit
	default:
		return m, nil
	}
}

func (m model) View() string {
	return "Hello"
}

//  docker run -d --name testbbt ubuntu sleep 9999
func main() {

	ch := make(chan bool)

	p := tea.NewProgram(model{ch: ch})

	go func() {
		<-time.After(time.Second)
		close(ch)
		p.Send(nil)
	}()

	if err := p.Start(); err != nil {
		fmt.Printf("Alas, there's been an error: %v", err)
		os.Exit(1)
	}

	cmd := exec.Command("bash", "-c", "docker exec -it testbbt bash")
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	cmd.Stdin = os.Stdin

	cmd.Run()
}
```

**Comment by meowgorithm** [comment]
Can confirm here as well with both Linux and macOS. The example is fine with Bubble Tea `v0.22.1` but in `v0.23.0`:

```
# linux
the input device is not a tty

# macOS
fork/exec /Applications/Docker.app/Contents/Resources/bin/com.docker.cli: bad file descriptor
```

So perhaps stdout's not being properly released.

**Comment by jaesolo** [comment]
Hi

**Comment by muesli** [comment]
Thanks for the detailed report! Fixed in `master` and the upcoming `v0.23.1`: we ended up closing `stdin` on `tea.Program` shutdown.

**Comment by meowgorithm** [comment]
@raphaelfff just a note that this is now fixed in release `v0.23.1`. Thanks again for the report!
